velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -742,17 +743,98 @@ class ParquetRowReader::Impl {
       uint64_t size,
       velox::VectorPtr& result,
       const dwio::common::Mutation* mutation) {
-    VELOX_DCHECK(!options_.getAppendRowNumberColumn());
     auto rowsToRead = nextReadSize(size);
     if (rowsToRead == kAtEnd) {
       return 0;
     }
     VELOX_DCHECK_GT(rowsToRead, 0);
-    columnReader_->next(rowsToRead, result, mutation);
+    if (!options_.getRowNumberColumnInfo().has_value()) {
+      columnReader_->next(rowsToRead, result, mutation);
+    } else {
+      readWithRowNumber(rowsToRead, result, mutation);
+    }
+
     currentRowInGroup_ += rowsToRead;
     return rowsToRead;
   }
 
+  void readWithRowNumber(
```

Comment:
The code is almost identical with DWRF, can you extract it to a common helper static method in `dwio::common::RowReader`?

Reply:
done